### PR TITLE
Added details to public static bool QueueUserWorkItem<TState> (Action…

### LINKED ISSUE #959: Missing documentation for QueueUserWorkItem<TState>(Action<TState>, TState, Boolean) .
There is no documentation for QueueUserWorkItem<TState>(Action<TState>, TState, Boolean) that explains what `bool preferLocal` does.

This was raised in an SO question [What is ThreadPool.QueueUserWorkItem preferLocal for?](https://stackoverflow.com/questions/52756327/what-is-threadpool-queueuserworkitem-preferlocal-for).

The [answer](https://stackoverflow.com/a/52756507/134204) shows that the method was added by [this pull request](https://github.com/dotnet/coreclr/pull/14214) fixing [this issue](https://github.com/dotnet/corefx/issues/12442). The issue's text and Damien's answer are a good starting point : 

> The issue is described as:

>ThreadPool.QueueUserWorkItem always queues to the global queue; however it would be good to have the option to be able to queue to the current threadpool thread's local queue when a threadpool thread queues extra work items.

...

>When the Thread Pool was first built, it only had a single queue of work to be done. However, when all of the Task goodness was being plumbed into the framework, they took the opportunity to introduce thread-local queues (and work stealing) alongside what was now renamed the global queue. It looks like this is clean-up work to allow specific access to these queues.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 21ac9edb-6827-475d-30d3-cfd3b52c3028
* Version Independent ID: e13ed79b-e212-1d7a-f939-6086b300a399
* Content: [ThreadPool.QueueUserWorkItem Method (System.Threading)](https://docs.microsoft.com/en-us/dotnet/api/system.threading.threadpool.queueuserworkitem?view=netcore-2.1#System_Threading_ThreadPool_QueueUserWorkItem__1_System_Action___0____0_System_Boolean_)
* Content Source: [xml/System.Threading/ThreadPool.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Threading/ThreadPool.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by peteraritchie**
…<TState> callBack, TState state, bool preferLocal)

## Summary

Added detail for public static bool QueueUserWorkItem<TState> (Action<TState> callBack, TState state, bool preferLocal)

Fixes #959

### CODE REVIEW

File: xml/System.Threading/ThreadPool.xml [hunk 1]
```diff
@@ -591,13 +591,14 @@
...
-        <param name="state">To be added.</param>
-        <param name="preferLocal">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="TState">The type of elements of <paramref name="state" />.</typeparam>
+        <param name="callBack">An <see cref="T:System.Action`1" /> representing the method to execute.</param>
+        <param name="state">An object containing data to be used by the method.</param>
+        <param name="preferLocal">Unlike other QueueUserWorkItem methods, prefer exeucting in the local thread pool and not the global thread pool.</param>
+        <summary>Queues a method for execution, and specifies an <see cref="T:System.Action`1" /> object containing data to be used by the method. The method executes when a thread pool thread becomes available.</summary>
```
Comment (rpetrusha) [inline xml/System.Threading/ThreadPool.xml]:
```suggestion
        <summary>Queues a method specified by an <see cref="T:System.Action`1" /> delegate for execution, and provides data to be used by the method. The method executes when a thread pool thread becomes available.</summary>
```
Delegate and data are separate argument; the delegate doesn't contain the data.

File: xml/System.Threading/ThreadPool.xml [hunk 2]
```diff
@@ -1267,13 +1268,13 @@
...
       <Docs>
-        <typeparam name="TState">To be added.</typeparam>
-        <param name="callBack">To be added.</param>
-        <param name="state">To be added.</param>
-        <param name="preferLocal">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <typeparam name="TState">The type of elements of <paramref name="state" />.</typeparam>
+        <param name="callBack">An <see cref="T:System.Action`1" /> representing the method to execute.</param>
```
Comment (rpetrusha) [inline xml/System.Threading/ThreadPool.xml]:
```suggestion
        <param name="callBack">A delegate representing the method to execute.</param>
```
The delegate type appears immediately above the description, so it's not necessary to duplicate it.

File: xml/System.Threading/ThreadPool.xml [hunk 3]
```diff
@@ -1267,13 +1268,13 @@
...
+        <param name="state">An object containing data to be used by the method.</param>
+        <param name="preferLocal">Unlike other QueueUserWorkItem methods, prefer exeucting in the local thread pool and not the global thread pool.</param>
```
Comment (rpetrusha) [inline xml/System.Threading/ThreadPool.xml]:
```suggestion
        <param name="preferLocal">`true` to execute in the local threadpool; `false` to execute in the global thread pool.</param>
```
The comment about other UnsafeQueueUserWorkItem methods can go in the Remarks section.

File: xml/System.Threading/ThreadPool.xml [hunk 4]
```diff
@@ -1267,13 +1268,13 @@
...
+        <summary>Queues a method for execution, and specifies an <see cref="T:System.Action`1" /> object containing data to be used by the method. The method executes when a thread pool thread becomes available.</summary>
```
Comment (rpetrusha) [inline xml/System.Threading/ThreadPool.xml]:
```suggestion
        <summary>Queues a method specified by an <see cref="T:System.Action`1" /> delegate for execution, and specifies an object containing data to be used by the method. The method executes when a thread pool thread becomes available.</summary>
```

**Comment by peteraritchie** [comment]
Thanks @rpetrusha great suggestions, I've merged and committed them to the PR.

**Comment by rpetrusha** [comment]
Thank you for making the additional changes, @peteraritchie. And again, thanks for documenting these two otherwise undocumented methods. We really appreciate it. I'll merge your PR now. It should be live on docs.microsoft.com in the next day or two.

**Pull request merged**